tests/31sync/06state.pl

```diff
@@ -258,7 +260,12 @@ sub wait_for_event_in_room {
                content => { "filler" => $_ },
                type    => "a.made.up.filler.type",
             )
-         } 0 .. 20 );
+         } 0 .. 19 );
+      })->then( sub {
+         matrix_send_room_message_and_wait_for_sync( $user, $room_id,
```

Comment:
We need to have a ``wait_for_sync`` so that before we call ``matrix_sync_again`` below. However invoking ``matrix_send_room_message_and_wait_for_sync`` in parallel is dangerous because we might skip over messages in a gapped sync causing ``wait_for_sync`` to wait forever.